### PR TITLE
Ae setup command

### PR DESCRIPTION
**Pull request by paetling**
review @jsingle  command to setup user initially

### CODE REVIEW

**Comment by jsingle** [comment]
Would be nice to be able to automatically set up some of these config keys. In our install script, we'll want to set the dusty-specs repo to our gc dusty specs repo.  Can't see how to automate that with the current setup...

File: dusty/cli/setup.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+"""Run this command once after installing dusty
+to configure dusty to run properly.
+
+Usage:
+  setup [--mac_username=<mac_username>] [--default_specs_repo=<specs_repo>] [--nginx_includes_dir=<nginx_dir>]
+
+Options:
+  --mac_username=<mac_username>         mac_username daemon will run under
```
Comment (jsingle) [inline dusty/cli/setup.py]: Really the daemon is just running select commands with this username

**Comment by jsingle** [comment]
:fireworks:

File: dusty/cli/setup.py (hunk 1 above)
Comment (paetling) [inline dusty/cli/setup.py]: what do you mean?

File: dusty/cli/__init__.py [hunk 2]
```diff
@@ -89,6 +92,9 @@ def main():
         sys.exit(1)
 
     configure_client_logging()
+    if not get_config_value('setup_has_run') and command != 'setup':
+        log_to_client('You must run `dusty setup` before you run any other commands')
+        sys.exit(0)
```
Comment (thieman) [inline dusty/cli/__init__.py]: @paetling This should be an error code

Comment (thieman) [inline dusty/cli/__init__.py]: I'ma send you a PR with some stuff

**Pull request merged**